app/src/main/java/org/simple/clinic/teleconsultlog/medicinefrequency/MedicineFrequencyEffectHandler.kt

```diff
@@ -0,0 +1,25 @@
+package org.simple.clinic.teleconsultlog.medicinefrequency
+
+import com.spotify.mobius.rx2.RxMobius
+import com.squareup.inject.assisted.Assisted
+import com.squareup.inject.assisted.AssistedInject
+import io.reactivex.ObservableTransformer
+import org.simple.clinic.util.scheduler.SchedulersProvider
+
+class MedicineFrequencyEffectHandler @AssistedInject constructor(
+    private val schedulersProvider: SchedulersProvider,
+    @Assisted private val uiActions: MedicineFrequencySheetUiActions
+) {
+
+  @AssistedInject.Factory
+  interface Factory {
+    fun create(uiActions: MedicineFrequencySheetUiActions): MedicineFrequencyEffectHandler
+  }
+
+  fun build(): ObservableTransformer<MedicineFrequencyEffect, MedicineFrequencyEvent> {
+    return RxMobius
+        .subtypeEffectHandler<MedicineFrequencyEffect, MedicineFrequencyEvent>()
+        .addConsumer(LoadDefaultMedicineFrequency::class.java, { uiActions.setMedicineFrequency(it.medicineFrequency) }, schedulersProvider.ui())
```

Comment:
The naming of it is a bit misleading because we are not actually loading the default medicine frequency. In an earlier comment, I meant changing the commit message to reflect setting medicine frequency.